rest-team.go

```diff
@@ -0,0 +1,286 @@
+package sdk
+
+/*
+   Copyright 2016 Alexander I.Grafov <grafov@gmail.com>
+   Copyright 2016-2019 The Grafana SDK authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+   ॐ तारे तुत्तारे तुरे स्व
+*/
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+)
+
+// SearchTeamsWithPaging search teams with paging.
+// query optional.  query value is contained in one of the name, login or email fields. Query values with spaces need to be url encoded e.g. query=Jane%20Doe
+// perpage optional. default 1000
+// page optional. default 1
+// http://docs.grafana.org/http_api/team/#search-teams
+// http://docs.grafana.org/http_api/team/#search-teams-with-paging
+//
+// Reflects GET /api/teams/search API call.
+func (r *Client) SearchTeamsWithPaging(ctx context.Context, query *string, perpage *int, page *int) (PageTeams, error) {
```

Comment:
Let's use this pattern for optional parameters: https://github.com/grafana-tools/sdk/blob/f7c763c3738c279f7e926c5a3bc8353b4080312b/rest-annotation.go#L93-L95 i.e. create a type for it and then call all functions in order. This is because these parameters are URL encoded, they are not part of the URL.